### PR TITLE
Database

### PR DESCRIPTION
**Pull request by jvikstrom**
### Test Plan
Checked rest endpoints /user (POST) and /user/{id} (GET). Also checked that the database actually saved the UserEntity. (Using the mongo console)

### Description
Adds connection to mongodb and a basic Dao to allow for most basic operations.
NOTE: There is an issue with this PR that makes AuthConfigFactory used by tomcat internally. The call in the Application class is a workaround for this.

It is probably related with intellij sharing classpath between all different maven projects. Kundera probably have a AuthConfigFactory that is not supported by Tomcat causing this error. This is not an issue for the moment with the workaround but we should probably clean the classpath (if that is what's wrong). But currently I think we should merge this and fix that after merging.

### CODE REVIEW

File: user-service/src/main/java/sjchat/users/UserServiceServer.java [hunk 1]
```diff
@@ -48,13 +52,22 @@ private void blockUntilShutdown() throws InterruptedException {
...
+    public UserService(){
+      super();
+      dao = new UserDaoImpl();
+    }
+
     @Override
     public void createUser(UserDataRequest req, StreamObserver<UserResponse> responseObserver) {
       Random random = new Random();
 
+      UserEntity entity = new UserEntity(Math.abs(random.nextInt(100)), req.getUsername());
```
Comment (alanihre) [inline user-service/src/main/java/sjchat/users/UserServiceServer.java]: Why is the user id set to a random number here? Can't it be generated by the database, maybe as an UUID instead?

File: database-service/src/main/java/sjchat/daos/UserDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * Created by Johan Vikström on 2017-04-01.
+ */
+public class UserDaoImpl extends DaoImpl implements UserDao{
+    public UserEntity findByUsername(String username){
+        Query q = em.createQuery("select u from UserEntity where u.username = :username", UserEntity.class);
+        q.setParameter("username", username);
+        List<UserEntity> users = q.getResultList();
+        if(users == null || users.isEmpty()){
+            return null;
+        }else{
```
Comment (alanihre) [inline database-service/src/main/java/sjchat/daos/UserDaoImpl.java]: Format code according to Googles Java code specification. This applies for other files as well.

**Comment by s4wny** [comment]
@jvikstrom Please fix the merge conflicts.

**Comment by s4wny** [comment]
@jvikstrom Also remove all .ideas files

**Pull request merged**